### ISSUE TITLE
Build can silently fail (without emitting an error) in low-memory environments

### ISSUE THREAD

**Issue by lostfictions** [zeit/next.js#6075]
# Bug report

## Describe the bug

My Next app is working fine locally -- both running in dev and with a production build -- but when building and deploying on a cloud server, I intermittently get an error during deploy:
```
       /code/node_modules/next/dist/server/next-server.js:753
               throw new Error("Could not find a valid build in the '".concat(this.distDir, "' directory! Try building your app with 'next build' before starting the server."));
                     ^
       Error: Could not find a valid build in the '/code/.next' directory! Try building your app with 'next build' before starting the server.
           at Server.readBuildId (/code/node_modules/next/dist/server/next-server.js:753:15)
           at new Server (/code/node_modules/next/dist/server/next-server.js:80:25)
           at Object.module.exports [as default] (/code/node_modules/next/dist/server/next.js:6:10)
           at Object.<anonymous> (/code/src/main.ts:87:21)
           at Module._compile (internal/modules/cjs/loader.js:721:30)
           at Object.Module._extensions..js (internal/modules/cjs/loader.js:732:10)
           at Module.load (internal/modules/cjs/loader.js:620:32)
           at tryModuleLoad (internal/modules/cjs/loader.js:560:12)
           at Function.Module._load (internal/modules/cjs/loader.js:552:3)
           at Function.Module.runMain (internal/modules/cjs/loader.js:774:12)
```

This is happening on a VM with 1gb ram without swap configured; in fact, when I added an extra gig of swap, the problem immediately goes away. #5575 might be relevant here.

## To Reproduce

- Create a memory-constrained VM
- Clone a reasonably complex Next example project
- Build it in the VM environment

## Expected behavior

The build emits the correct files.

## System information
 - OS: Ubuntu 16.04
 - Version of Next.js: 7.0.2

**Comment by timneutkens** [comment]
Please see https://github.com/zeit/next.js/issues/5575#issuecomment-455155535 it's an upstream webpack issue.

**Comment by timneutkens** [comment]
Btw if it runs out of memory it's not going to silently fail, it will either exit with status 137 or a giant error message from Node.js saying you ran out of heap memory.

**Comment by lostfictions** [comment]
I wonder if it's an error with the suggested config for `next-typescript` [using the fork-ts-checker-webpack-plugin](https://github.com/zeit/next-plugins/tree/master/packages/next-typescript#type-checking), then -- what seems to happen is that, where I typically get a message like this:
```
$ next build
[4:16:42 PM] Compiling client
[4:16:42 PM] Compiling server
> Using external babel configuration
> Location: "/code/.babelrc"
[4:17:18 PM] Compiled server in 36s
[4:17:19 PM] Compiled client in 36s
```

when something goes wrong, only the client shows up as being successfully compiled, with no message about the server whatsoever.

**Comment by timneutkens** [comment]
I think this is solved in next@canary because we've removed the childprocess spawning.

**Comment by nnennaude** [comment]
This happened to me when I removed a file that one of my pages (in the /pages folder) happened to be referencing.

Either restore that file, or remove the file that is referencing the missing file.

**Comment by pietro-tavares** [comment]
I had this problem today with my Next.js app n Azure Container Instance. My build was failing silently (process killed/SIGTERMed) causing the container to restart.

Setting 2GB of RAM fixed the issue for me.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
